applications/OptimizationApplication/python_scripts/algorithms/algorithm_steepest_descent.py

```diff
@@ -66,69 +67,65 @@ def GetMinimumBufferSize(self) -> int:
     def Check(self):
         pass
 
+    @time_decorator()
     def Initialize(self):
         self.converged = False
         self.__obj_val = None
         self.__objective.Initialize()
         self.__objective.Check()
         self.master_control.Initialize()
-        self.__control_field = self.master_control.GetControlField() # GetInitialControlFields() later
+        self.__control_field = self.master_control.GetControlField()
+        self.algorithm_data = ComponentDataView("algorithm", self._optimization_problem)
 
     def Finalize(self):
-        pass
+        self.__objective.Finalize()
+        self.master_control.Finalize()
 
+    @time_decorator()
     def ComputeSearchDirection(self, obj_grad) -> KratosOA.CollectiveExpression:
-        return obj_grad * -1.0
+        search_direction = obj_grad * -1.0
+        self.algorithm_data.GetBufferedData()["search_direction"] = search_direction
+        return search_direction
+
+    @time_decorator()
+    def UpdateControlField(self, alpha) -> KratosOA.CollectiveExpression:
+        update = self.algorithm_data.GetBufferedData()["search_direction"] * alpha
+        self.__control_field += update
+        self.algorithm_data.GetBufferedData()["parameter_update"] = update
+        self.algorithm_data.GetBufferedData()["control_field"] = self.__control_field
 
     def GetCurrentObjValue(self) -> float:
         return self.__obj_val
 
     def GetCurrentControlField(self):
         return self.__control_field
 
-    def SolveOptimizationProblem(self) -> bool:
-        self.Initialize()
-        with TimeLogger("Solve Optimization problem", "Start", "End"):
-            self.Solve()
-        return self.converged
-
+    @time_decorator()
     def Solve(self):
-        algorithm_data = ComponentDataView("algorithm", self._optimization_problem)
         while not self.converged:
-            print("")
-            with TimeLogger("Optimization", f" Start Iteration {self._optimization_problem.GetStep()}", f"End Iteration {self._optimization_problem.GetStep()}"):
+            with OptimizationAlgorithmTimeLogger("AlgorithmSteepestDescent",self._optimization_problem.GetStep()):
+                self.__obj_val = self.__objective.CalculateStandardizedValue(self.__control_field)
+                obj_info = self.__objective.GetInfo()
+                self.algorithm_data.GetBufferedData()["std_obj_value"] = obj_info["value"]
+                self.algorithm_data.GetBufferedData()["rel_obj[%]"] = obj_info["rel_change [%]"]
+                if "abs_change [%]" in obj_info:
+                    self.algorithm_data.GetBufferedData()["abs_obj[%]"] = obj_info["abs_change [%]"]
```

Comment:
I think this is always true ryt? Hence I would remove the if block.

Reply:
If init value is 0, you can't compute abs change in % as reference is 0.